### PR TITLE
fix: split MPI communicator to exclude non DR nodes from the reduce

### PR DESCRIPTION
**Pull request by Thomas-Ulrich**
sorry about that

### CODE REVIEW

File: src/ResultWriter/magnitude_output.f90 [hunk 1]
```diff
@@ -79,16 +79,15 @@ SUBROUTINE magnitude_output(MaterialVal,DISC,MESH,MPI,IO)
     !-------------------------------------------------------------------------!
     ! Local variable declaration                                              !
     INTEGER                         :: iElem,iSide,nSide,iFace
-    INTEGER                         :: stat, UNIT_MAG, iErr
+    INTEGER                         :: stat, UNIT_MAG, iErr, DR_comm
```
Comment (uphoffc) [inline src/ResultWriter/magnitude_output.f90:82]: Should be TYPE(MPI_Comm) :: DR_comm

File: src/Solver/calc_seissol.f90 [hunk 2]
```diff
@@ -104,6 +104,7 @@ SUBROUTINE calc_SeisSol(time,timestep,pvar,cvar,EQN,MESH,DISC,SOURCE,BND,IC, &
     INTEGER                       :: plotNr, minl(1), iter
     INTEGER                       :: iElem, iSide, i, iObject, MPIIndex
     INTEGER(KIND=4)               :: now(3)
+    integer                       :: DR_comm                        !< dynamic rupture communicator
```
Comment (uphoffc) [inline src/Solver/calc_seissol.f90:107]: Should be TYPE(MPI_Comm) :: DR_comm

File: src/Solver/calc_seissol.f90 [hunk 3]
```diff
@@ -215,8 +216,12 @@ SUBROUTINE calc_SeisSol(time,timestep,pvar,cvar,EQN,MESH,DISC,SOURCE,BND,IC, &
 #endif
     ENDIF
 
+#ifdef USE_MPI
+    CALL MPI_Comm_split(MPI%commWorld, EQN%DR, 1, DR_comm, iErr)
```
Comment (uphoffc) [inline src/Solver/calc_seissol.f90:220]: This will only work if rank 0 has dynamic rupture faces!

**Review by uphoffc — CHANGES_REQUESTED** [review]
This will only work if EQN%DR = 1 for rank 0. Otherwise, rank 0 ends up in the other group and the magnitude will be zero.

You might just remove EQN%DR.EQ.1 in Line 224 instead of splitting the communicator. Then magnitude_output is called by all processes.

**Comment by Thomas-Ulrich** [comment]
Thanks for the review!
Could it work like that?
```
diff --git a/src/ResultWriter/magnitude_output.f90 b/src/ResultWriter/magnitude_output.f90
index bfc096b8..4c528004 100644
--- a/src/ResultWriter/magnitude_output.f90
+++ b/src/ResultWriter/magnitude_output.f90
@@ -76,10 +76,11 @@ CONTAINS
     TYPE(tUnstructMesh)             :: MESH
     TYPE(tMPI)                      :: MPI
     TYPE(tInputOutput)              :: IO
+    TYPE(MPI_Comm)                  :: DR_comm
     !-------------------------------------------------------------------------!
     ! Local variable declaration                                              !
     INTEGER                         :: iElem,iSide,nSide,iFace
-    INTEGER                         :: stat, UNIT_MAG, iErr, DR_comm
+    INTEGER                         :: stat, UNIT_MAG, iErr, rankDR
     REAL                            :: magnitude, magnitude0
     REAL                            :: MaterialVal(:,:)
     LOGICAL                         :: exist
@@ -105,10 +106,12 @@ CONTAINS
     ENDDO
 #ifdef PARALLEL
     CALL MPI_REDUCE(magnitude,magnitude0,1,MPI%MPI_AUTO_REAL,MPI_SUM,0, DR_comm,iErr)
+    CALL MPI_Comm_rank(DR_comm, rankDR, iErr)
 #else
     magnitude0 = magnitude
+    rankDR = 0
 #endif
-    IF (MPI%myrank.EQ.0) THEN
+    IF (rankDR.EQ.0) THEN

         WRITE(MAG_FILE, '(a,a4,a4)') TRIM(IO%OutputFile),'-MAG','.dat'
         UNIT_MAG = 299875
diff --git a/src/Solver/calc_seissol.f90 b/src/Solver/calc_seissol.f90
index 08ceabfd..2f7f0435 100644
--- a/src/Solver/calc_seissol.f90
+++ b/src/Solver/calc_seissol.f90
@@ -94,6 +94,7 @@ CONTAINS
     TYPE (tUnstructOptionalFields):: OptionalFields
     TYPE (tInputOutput)           :: IO
     TYPE (tMPI)                   :: MPI
+    TYPE(MPI_Comm)                :: DR_comm !< dynamic rupture communicator

     REAL                          :: time                 ! current time
     INTEGER                       :: timestep             ! index of time step
@@ -104,7 +105,6 @@ CONTAINS
     INTEGER                       :: plotNr, minl(1), iter
     INTEGER                       :: iElem, iSide, i, iObject, MPIIndex
     INTEGER(KIND=4)               :: now(3)
-    integer                       :: DR_comm                        !< dynamic rupture communicator
     REAL                          :: tol, InitBegin, InitEnd, LoopBegin, LoopEnd
     REAL                          :: MPITime, MPIdt, dt
     REAL                          :: CPUTime

```
The TYPE(MPI_Comm) does not work for me:
```
/hppfs/work/pr45fi/di73yeq4/SeisSol/src/ResultWriter/magnitude_output.f90(79): error #6457: This derived type name has not been declared.   [MPI_COMM]
    TYPE(MPI_Comm)                  :: DR_comm
```

**Comment by uphoffc** [comment]
Yes that should work.

Maybe TYPE(MPI_COMM_WORLD) ?

**Comment by Thomas-Ulrich** [comment]
> Yes that should work.
> 
> Maybe TYPE(MPI_COMM_WORLD) ?

does not work.
But here it is declared as integer:
https://github.com/SeisSol/SeisSol/blob/master/src/Numerical_aux/typesdef.f90#L739

**Comment by uphoffc** [comment]
ok whatever, then just use integer

**Review by uphoffc — APPROVED** [review]

**Comment by Thomas-Ulrich** [comment]
I can't merge.

**Pull request merged**